sycl/doc/extensions/Matrix/dpcpp-joint-matrix.asciidoc

```diff
@@ -0,0 +1,311 @@
+value to determine which of the extension's APIs the implementation supports.
+
+[frame="none",options="header"]
+|======================
+|Value |Description
+|1     |Initial extension implementation on AMX.  Base features are supported.
+|======================
+
+## New `joint_matrix` class
+We introduce a new class called `joint_matrix`. The user needs to specify the type of the elements, shape, the memory layout, and the memory scope of the matrix. This results into the following description:
+
+```c++
+namespace sycl::ext::intel::experimental::matrix {
+template <typename Group, typename T, size_t Rows=sycl::dynamic_extent, size_t Cols=sycl::dynamic_extent, matrix_layout Layout = matrix_layout::row_major>
+struct joint_matrix {
+    joint_matrix(Group g) {}
+};
+}
+```
+
+
+#### Memory Scope
+In this experimental API version, we used the terminology of `joint_matrix` instead of plain `matrix` to emphasis that the matrix is shared among a group of work items and is not private to each work item. The memory scope is added as an additional template parameter and is also part of the constructor arguments.
+
+IMPORTANT: In the current implementation, only the subgroup scope is supported
+
+When the group is a `sycl::sub_group`, a matrix is declared as follows:
+
+```c++
+joint_matrix<sub_group, int8_t, tM, tN> tA(sg); 
+```
+
+#### Shape
+The same class `joint_matrix` should handle both cases where sizes are constant (GPU case) and when sizes are variables (CPU case). Note that a AMX 2d tile register permits sizes up to 1024 (16rowsx64cols) bytes. The ability to define only one interface for both makes it possible to give the user a way to make use of the flexibility introduced by the CPU but at the same time save resources on the GPU. We use `sycl::dynamic_extent`  to differentiate between static and dynamic sizes.
+
+IMPORTANT: In the current implementation, only the static extent is supported
+
+
+#### Layout
+//Layout is necessary on the type to be able to calculate the physical offset if the user needs to access a single entry for some purpose.
+Besides row major and column major layouts, `matrix_layout` is flexible enough to introduce customed layouts such as symmetric or tiled layouts.
+	
+```c++
+namespace sycl::ext::intel::experimental::matrix {
+enum class matrix_layout {
+  row_major,
+  col_major,
+  packed_a,
+  packed_b
+};
+}
+```
+
+AMX hardware requires both A and B to be in VNNI or 32 bits packed layout. So the matrices are always packed in AMX. If we multiply matrices A (M, K) and B (K, N) into a matrix C (M, N). The logical sizes are M, K, N. However, the packed shape for B used by the implementation is different. For bf16 type for instance, the shape of AMX tile B is (K/2, N*2). The user must provide the information of `packed_b` layout to make the implementation allocate the right shape.  The layout information for AMX should be specified in user code as follows: 
```

Comment:
>> However, the B matrix must be stored in memory using the VNNI format, which is described below.

I don't prefer mentioning memory here because we are talking about the matrix that will be allocated in a register.
I will reword this to:

"AMX hardware requires B matrix to be in VNNI or 32 bits packed layout. If we multiply matrices A (M, K) and B (K, N) into a matrix C (M, N). The logical sizes are M, K, N. However, the packed shape for B tile uses the VNNI format, which is described below. The user must provide the information of `packed_b` layout to make the implementation allocate the right shape. "

What do you think?